<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>generateDialog</class>
 <widget class="QDialog" name="generateDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>288</width>
    <height>386</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Генерация новой конфигурации системы (старая удалится)</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Высота</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="height">
       <property name="value">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Ширина</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="width">
       <property name="value">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QDoubleSpinBox" name="mLength">
       <property name="minimum">
        <double>0.010000000000000</double>
       </property>
       <property name="maximum">
        <double>10000.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>50.000000000000000</double>
       </property>
       <property name="value">
        <double>200.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>|m| (% от радиуса)</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Радиус частицы</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QDoubleSpinBox" name="radius">
       <property name="minimum">
        <double>0.010000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
       <property name="value">
        <double>0.500000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QToolBox" name="genType">
     <property name="accessibleName">
      <string/>
     </property>
     <widget class="QWidget" name="genRandomConcentration" native="true">
      <attribute name="label">
       <string>Случайно (по плотности)</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_3">
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Плотность</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="partDestiny">
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>0.500000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="genRandomCount" native="true">
      <attribute name="label">
       <string>Случайно (по количеству)</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Количество</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="partCount">
         <property name="value">
          <number>10</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="genByChain" native="true">
      <attribute name="label">
       <string>Цепочка</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Расстояние между частицами</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="partSpacing">
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.500000000000000</double>
         </property>
         <property name="value">
          <double>0.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QFormLayout" name="formLayout_5">
      <item row="0" column="0">
       <widget class="QLabel" name="particleCountLabel">
        <property name="text">
         <string>Количество = </string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="particleCount">
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="particleConcentrationLabel">
        <property name="text">
         <string>Концентрация =</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="particleConcentration">
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="squareLabel">
        <property name="text">
         <string>Площадь = </string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="square">
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>buttonBox</tabstop>
  <tabstop>height</tabstop>
  <tabstop>width</tabstop>
  <tabstop>radius</tabstop>
  <tabstop>mLength</tabstop>
  <tabstop>partDestiny</tabstop>
  <tabstop>partCount</tabstop>
  <tabstop>partSpacing</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>height</sender>
   <signal>valueChanged(int)</signal>
   <receiver>generateDialog</receiver>
   <slot>recalcInfoLabels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>width</sender>
   <signal>valueChanged(int)</signal>
   <receiver>generateDialog</receiver>
   <slot>recalcInfoLabels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radius</sender>
   <signal>valueChanged(double)</signal>
   <receiver>generateDialog</receiver>
   <slot>recalcInfoLabels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mLength</sender>
   <signal>valueChanged(double)</signal>
   <receiver>generateDialog</receiver>
   <slot>recalcInfoLabels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>partDestiny</sender>
   <signal>valueChanged(double)</signal>
   <receiver>generateDialog</receiver>
   <slot>recalcInfoLabels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>partCount</sender>
   <signal>valueChanged(int)</signal>
   <receiver>generateDialog</receiver>
   <slot>recalcInfoLabels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>176</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>partSpacing</sender>
   <signal>valueChanged(double)</signal>
   <receiver>generateDialog</receiver>
   <slot>recalcInfoLabels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>genType</sender>
   <signal>currentChanged(int)</signal>
   <receiver>generateDialog</receiver>
   <slot>recalcInfoLabels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>generateDest()</slot>
  <slot>generteCount()</slot>
  <slot>recalcInfoLabels()</slot>
 </slots>
</ui>
